Refactor docs for database dev services (#25403)
quarkusio/quarkus: docs/src/main/asciidoc/databases-dev-services.adoc

By default, Quarkus uses the default image for the current version of Testcontainers, if you attempt to start Quarkus the resulting failure will tell you the exact image name in use for you to add to the file.

An example file is shown below:

.src/main/resources/container-license-acceptance.txt
----
ibmcom/db2:11.5.0.0a
mcr.microsoft.com/mssql/server:2017-CU12
----

== Database Vendor Specific Configuration

All services based on containers are run using Testcontainers but Quarkus is not using the Testcontainers JDBC driver.
Thus, even though extra JDBC URL properties can be set in your `application.properties` file, specific properties supported by the Testcontainers JDBC driver such as `TC_INITSCRIPT`, `TC_INITFUNCTION`, `TC_DAEMON`, `TC_TMPFS` are not supported.

Quarkus can support *specific* properties sent to the container itself though, e.g. this is the case for `TC_MY_CNF` which allows to override the MariaDB/MySQL configuration file.

Overriding the MariaDB/MySQL configuration would be done as follows:

[source,properties]
----
quarkus.datasource.devservices.container-properties.TC_MY_CNF=testcontainers/mysql-conf
----

> PostgreSQL docker images can be configured via docker command (see https://github.com/docker-library/docs/blob/master/postgres/README.md#database-configuration[the official docu]).

Reviewer: ```suggestion
PostgreSQL Docker images can be configured via a Docker command (see https://github.com/docker-library/docs/blob/master/postgres/README.md#database-configuration[the official documentation]).
```

> For the PostgreSQL Dev Service the command can be set like this:

Reviewer: ```suggestion
For instance, a command could be set like this:
```

Author: I took out the whole part. Will be added back during https://github.com/quarkusio/quarkus/pull/25289

[source,properties]
----
quarkus.datasource.devservices.container-properties.CMD=postgres -c wal_level=logical
----

> This support is database specific and needs to be implemented in each dev service specifically.

Reviewer: This part will need adjustments once https://github.com/quarkusio/quarkus/pull/25289 is taken care of.

Author: I took out the whole part. Will be added back during https://github.com/quarkusio/quarkus/pull/25289
